setlexsem/generate/generate_sets.py

```diff
@@ -44,28 +45,69 @@ def get_parser():
     return parser
 
 
+def astype_set(raw_input):
+    if type(raw_input) == str:
+        raw_input = raw_input.strip()
+        set_out = ast.literal_eval(raw_input)
+    else:
+        set_out = raw_input
+    return set_out
+
+
+def parse_set_pair(raw_set_a: str, raw_set_b: str) -> Tuple[set, set]:
+    """Parse string representations of sets into Python sets"""
+    try:
+        set_a = astype_set(raw_set_a)
+        set_b = astype_set(raw_set_b)
+        return set_a, set_b
+    except (ValueError, SyntaxError) as e:
+        logger.error(f"Failed to parse sets: {e}")
+        raise
+
+
+def generate_set_pair(sampler: Union[Iterable, callable]) -> Tuple[set, set]:
+    """Generate a pair of sets from either an iterable or callable sampler"""
```

Comment:
Can you add more to this doc string that explains, to the user, the motivation for this function?